gax-java/gax/src/test/java/com/google/api/gax/tracing/OpentelemetryMetricsRecorderTest.java

```diff
@@ -0,0 +1,271 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.tracing;
+
+import static org.junit.Assert.assertThrows;
+
+import com.google.api.gax.core.GaxProperties;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.truth.Truth;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.data.Data;
+import io.opentelemetry.sdk.metrics.data.HistogramPointData;
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.PointData;
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricExporter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class OpentelemetryMetricsRecorderTest {
+  private OpentelemetryMetricsRecorder otelMetricsRecorder;
+  private static InMemoryMetricExporter exporter;
+  private static PeriodicMetricReader periodicMetricReader;
+
+  @BeforeClass
+  public static void setUpBeforeClass() {
+    exporter = InMemoryMetricExporter.create();
+    periodicMetricReader = PeriodicMetricReader.create(exporter);
+  }
+
+  @Before
+  public void setUp() {
+    SdkMeterProvider sdkMeterProvider =
+        SdkMeterProvider.builder().registerMetricReader(periodicMetricReader).build();
+    OpenTelemetry openTelemetry =
+        OpenTelemetrySdk.builder().setMeterProvider(sdkMeterProvider).build();
+
+    // Meter Creation
+    Meter meter =
+        openTelemetry
+            .meterBuilder("OtelMetricsRecorderTest")
+            .setInstrumentationVersion(GaxProperties.getGaxVersion())
+            .build();
+
+    otelMetricsRecorder = new OpentelemetryMetricsRecorder(meter);
+  }
```

Comment:
On the fence about using an InMemory(Reader|Exporter) here. Might not be needed since I believe this should be covered in showcase tests, but provides the extra benefit of testing each individual metric in isolation (multiple invocations + each invocation records attributes)..

Reply:
Revert to not use InMemory for this unit test.